src/UI/Components/Diagram/paper/paper.ts

```diff
@@ -0,0 +1,284 @@
+      // We don't want a Halo if cellView is a Link or is a representation of an already existing instance that has strict_modifier set to false
+      if (
+        cellView.model instanceof dia.Link ||
+        cellView.model.get("isBlockedFromEditing")
+      )
+        return;
+      if (cellView.model.get("isBlockedFromEditing") || !editable) return;
+
+      const halo = createHalo(
+        graph,
+        this.paper,
+        cellView,
+        connectionRules,
+        updateInstancesToSend,
+      );
+
+      halo.render();
+    });
+
+    this.paper.on("link:mouseenter", (linkView) => {
+      const source = linkView.model.source();
+      const target = linkView.model.target();
+
+      const sourceCell = graph.getCell(
+        source.id as dia.Cell.ID,
+      ) as ServiceEntityBlock;
+      const targetCell = graph.getCell(
+        target.id as dia.Cell.ID,
+      ) as ServiceEntityBlock;
+      if (!(sourceCell.getName()[0] === "_")) {
+        linkView.model.appendLabel({
+          attrs: {
+            rect: {
+              fill: "none",
+            },
+            text: {
+              text: sourceCell.getName(),
+              autoOrient: "target",
+              class: "joint-label-text",
+            },
+          },
+          position: {
+            distance: 1,
+          },
+        });
+      }
+      if (!(targetCell.getName()[0] === "_")) {
+        linkView.model.appendLabel({
+          attrs: {
+            rect: {
+              fill: "none",
+            },
+            text: {
+              text: targetCell.getName(),
+              autoOrient: "source",
+              class: "joint-label-text",
+            },
+          },
+          position: {
+            distance: 0,
+          },
+        });
+      }
+      if (linkView.model.get("isBlockedFromEditing") || !editable) return;
+      showLinkTools(
+        this.paper,
+        graph,
+        linkView,
+        updateInstancesToSend,
+        connectionRules,
+      );
+    });
+
+    this.paper.on("link:mouseleave", (linkView: dia.LinkView) => {
+      linkView.removeTools();
+      linkView.model.labels([]);
+    });
+
+    this.paper.on("link:connect", (linkView: dia.LinkView) => {
+      //only id values are stored in the linkView
+      const source = linkView.model.source();
+      const target = linkView.model.target();
+
+      const sourceCell = graph.getCell(
+        source.id as dia.Cell.ID,
+      ) as ServiceEntityBlock;
+      const targetCell = graph.getCell(
+        target.id as dia.Cell.ID,
+      ) as ServiceEntityBlock;
+
+      /**
+       * Function that checks if cell that we are connecting  to is being the one storing information about said connection.
+       * @param elementCell cell that we checking
+       * @param connectingCell cell that is being connected to elementCell
+       * @returns boolean whether connections was set
+       */
+      const wasConnectionDataAssigned = (
```

Comment:
A promise? I don't feel it. If you don't like it I can extract assigement into separate function

Reply:
In essence, it's what you are doing, whether you like it or not 🙃 
You perform a task and if it succeeds you return it's success state/error state, you picked a boolean for that, but it's the same. And then you do something with that success state. For me, that's called a promise.

Reply:
For me, to make code clear and readable, a method can either:
- Perform computations and return a new value
- Perform an action
- Assert whether something matches a certain requirement. 

But not a mix of everything.

Reply:
conditionals aren't promises, here there is an attempt to assign data about the given connection with the returned result in case of boolean if the connection data was assigned. I'll link this comment it to the right sidebar task and handle it there

Reply:
What you have here, is not a simple conditional. And an attempt to assign data to then do something, is very close to the definition of a promise. 

`assignData.then(your logic)` 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then

There's nothing wrong with needing promises. Or however you wish to call them.

Reply:
removed the boolean and just call the functions for both cells.  it simplified the function to one purpose only and there is no need to extract or split function more(at least in my opinion) and conditionals won't allow to double assign the information